src/Domain/SegmentId.hpp

```diff
@@ -121,7 +121,12 @@ class SegmentId {
 
   SegmentId(size_t block_id, size_t refinement_level, size_t index) noexcept;
   size_t block_id() const noexcept { return block_id_; }
-  void set_block_id(const size_t block_id) noexcept { block_id_ = block_id; }
+  void set_block_id(const size_t block_id) noexcept {
+    ASSERT(block_id < two_to_the(block_id_bits),
```

Comment:
I'm fine with the `ASSERT` but I don't see how the release code path changes and how this fixes a bug. I can see that it helps catch other bugs, but this code functionally still does `block_id_ = block_id`. The commit message seems misleading